community/codegen/src/main/java/org/neo4j/codegen/source/SourceCode.java

```diff
@@ -61,6 +63,24 @@ public String toString()
         {
             return "SourceCode";
         }
+
+        private String fullClasspath()
+        {
+            StringBuilder classpath = new StringBuilder( System.getProperty( "java.class.path" ) );
+
+            ClassLoader classLoader = SourceCode.class.getClassLoader();
+            if ( classLoader instanceof URLClassLoader )
+            {
+                URL[] classpathUrls = ((URLClassLoader) classLoader).getURLs();
+                for ( URL classpathUrl : classpathUrls )
+                {
+                    classpath.append( File.pathSeparator );
+                    classpath.append( classpathUrl.getPath() );
+                }
+            }
```

Comment:
It seems strange to *both* fetch the `java.class.path` property *and* go through the class loader to add the paths.

It seems like either we should do one or the other, if we do both, we should probably convert it to an (ordered) set of paths before converting it to a string, so that we don't add things twice. Probably we should also canonicalise the paths.